src/command_modules/azure-cli-network/azure/cli/command_modules/network/custom.py

```diff
@@ -820,6 +828,72 @@ def update_asg(instance, tags=None):
 # endregion
 
 
+# region DdosProtectionPlans
+def create_ddos_plan(cmd, resource_group_name, ddos_plan_name, location=None, tags=None, vnets=None):
+    from azure.cli.core.commands import LongRunningOperation
+
+    ddos_client = network_client_factory(cmd.cli_ctx).ddos_protection_plans
+    DdosProtectionPlan, SubResource = cmd.get_models('DdosProtectionPlan', 'SubResource')
+    plan = DdosProtectionPlan(
+        name=ddos_plan_name,
+        location=location,
+        tags=tags
+    )
+    if not vnets:
+        # if no VNETs can do a simple PUT
+        return ddos_client.create_or_update(resource_group_name, ddos_plan_name, plan)
+
+    # if VNETs specified, have to create the protection plan and then add the VNETs
+    plan_id = LongRunningOperation(cmd.cli_ctx)(
+        ddos_client.create_or_update(resource_group_name, ddos_plan_name, plan)).id
+
+    logger.info('Attempting to attach VNets to newly created DDoS protection plan.')
+    for subresource in vnets:
+        vnet_client = network_client_factory(cmd.cli_ctx).virtual_networks
+        id_parts = parse_resource_id(subresource.id)
+        vnet = vnet_client.get(id_parts['resource_group'], id_parts['name'])
+        vnet.ddos_protection_plan = SubResource(id=plan_id)
+        vnet_client.create_or_update(id_parts['resource_group'], id_parts['name'], vnet)
+    return ddos_client.get(resource_group_name, ddos_plan_name)
+
+
+def update_ddos_plan(cmd, instance, tags=None, vnets=None):
+    SubResource = cmd.get_models('SubResource')
+
+    if tags is not None:
+        instance.tags = tags
+    if vnets == "":
+        vnets = []
+    if vnets is not None:
+        logger.info('Attempting to update the VNets attached to the DDoS protection plan.')
+        vnet_ids = [x.id for x in vnets]
+        existing_vnets = [x.id for x in instance.virtual_networks or []]
+        vnets_to_remove = [x for x in existing_vnets if x not in vnet_ids]
+        vnets_to_add = [x for x in vnet_ids if x not in existing_vnets]
```

Comment:
In case you like  you can leverage set to simplify and also improve the variable naming:
```python
   vnet_ids = set([x.id for x in vnets])
   existing_vnet_ids = set([x.id for x in instance.virtual_networks or []])
   for vnet_id in vnet_ids.difference(existing_vnet_ids):
         # do add
   for vnet_id in existing_vnet_ids.difference(vnet_ids):
         # do remove
```